doc/sphinx/proof-engine/proof-handling.rst

```diff
@@ -1,581 +1,584 @@
-  terms}\index{Proof term}.
+terms of λ-calculus, known as the *Curry-Howard isomorphism* [[How80]_,
+[Bar81]_, [Gir89]_, [Hue88]_ ], |Coq|
+stores proofs as terms of |Cic|. Those terms
+are called *proof terms*.
 
-\ErrMsg When one attempts to use a proof editing command out of the
-proof editing mode, \Coq~ raises the error message : \errindex{No focused
-  proof}.
 
-\section{Switching on/off the proof editing mode}
+Error message: When one attempts to use a proof editing command out of
+the proof editing mode, Coq raises the error message : ``No focused
+proof``.
+
+Switching on/off the proof editing mode
+-------------------------------------------
 
 The proof editing mode is entered by asserting a statement, which
 typically is the assertion of a theorem:
 
-\begin{quote}
-{\tt Theorem {\ident} \zeroone{\binders} : {\form}.\comindex{Theorem}
-\label{Theorem}}
-\end{quote}
+.. cmd:: Theorem @ident [@binders] : @form.
 
-The list of assertion commands is given in
-Section~\ref{Assertions}. The command {\tt Goal} can also be used.
+The list of assertion commands is given in Section :ref:TODO-assertions_and_proof`. The
+command ``Goal`` can also be used.
 
-\subsection[Goal {\form}.]{\tt Goal {\form}.\comindex{Goal}\label{Goal}}
+.. cmd:: Goal @form.
 
 This is intended for quick assertion of statements, without knowing in
 advance which name to give to the assertion, typically for quick
 testing of the provability of a statement. If the proof of the
 statement is eventually completed and validated, the statement is then
-bound to the name {\tt Unnamed\_thm} (or a variant of this name not
-already used for another statement).
+bound to the name ``Unnamed_thm`` (or a variant of this name not already
+used for another statement).
+
+.. cmd:: Qed.
 
-\subsection[\tt Qed.]{\tt Qed.\comindex{Qed}\label{Qed}}
 This command is available in interactive editing proof mode when the
-proof is completed.  Then {\tt Qed} extracts a proof term from the
-proof script, switches back to {\Coq} top-level and attaches the
-extracted proof term to the declared name of the original goal. This
-name is added to the environment as an {\tt Opaque} constant.
-
-\begin{ErrMsgs}
-\item \errindex{Attempt to save an incomplete proof}
-%\item \ident\ \errindex{already exists}\\
-%  The implicit name is already defined. You have then to provide
-%  explicitly a new name (see variant 3 below).
-\item Sometimes an error occurs when building the proof term,
-because tactics do not enforce completely the term construction
+proof is completed. Then ``Qed`` extracts a proof term from the proof
+script, switches back to Coq top-level and attaches the extracted
+proof term to the declared name of the original goal. This name is
+added to the environment as an ``Opaque`` constant.
+
+
+.. exn:: Attempt to save an incomplete proof
+
+Sometimes an error occurs when building the proof term, because
+tactics do not enforce completely the term construction
```

Comment:
This is not the description corresponding to the previous error message. It was more clear in the LaTeX version because there was a bullet point for the error message and another one for this paragraph.

Reply:
Not sure what I should do about it.

Reply:
Add `..remark::`?